src/systems/diff_context.C

```diff
@@ -18,6 +18,8 @@
 
 #include "libmesh/diff_context.h"
 #include "libmesh/diff_system.h"
+#include "libmesh/fem_system.h"
```

Comment:
You don't need this.  Cast to DiffSystem and you can still get the TimeSolver out.  We still might implement FDMSystem one of these days!